### PR TITLE
#8233 documenting git commands to synchronise an in progress PR with …

### LINKED ISSUE #8233: Feature Request/Idea: Describe the git commands to synchronise an in progress PR with the current development branch
**Overview of the Feature Request**
Describe the git commands which required to update an in progress PR with the current development branch. This should be available in the Developer Guide's Version control page.

**What kind of user is the feature intended for?**
Dataverse developers

**What inspired the request?**
Sometimes I needed to check other Git documentation what should be the requied git commands. It might be helpful for other developers.

**What existing behavior do you want changed?**
Extending the documentation, no behavior change.

**Any brand new behavior do you want to add to Dataverse?**
No.

**Any related open or closed issues to this feature request?**
No.

### PR DESCRIPTION
**Pull request by pkiraly**
…the current development branch

**What this PR does / why we need it**: Adding some extra documentation of git commands for Dataverse developers

**Which issue(s) this PR closes**:

Closes #8233 

**Special notes for your reviewer**:
This is a documentation change. It adds two normal sentences, and a code block with Git commands and comments

**Suggestions on how to test this**:
Build the documentation, and check `doc/sphinx-guides/build/html/developers/version-control.html` file.

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:
No.

**Is there a release notes update needed for this change?**:
No.

**Additional documentation**:
No.

### CODE REVIEW

**Review by pdurbin — COMMENTED** [review]
I wasn't aware of similar content having been added (also by @pkiraly) in pull request #6598 a year and a half ago.

Overall all, the existing and new content look fine. A typo ("banch") appears twice and should be fixed but otherwise I mostly just have comments.

First, I sometimes question how we (and perhaps I mean "I", depending on what `git blame` reveals) have oriented contributors toward cloning a fork of the repo (in the dev environment page) rather than cloning the main repo. I say this because although it's the documented procedure, no one on the core team does this so we don't have to worry about keeping our "develop" branches up-to-date, for example. Péter's examples introduce the idea of setting up a remote called "upstream". This is great (assuming we stick with the "clone your fork" model) but perhaps it should be introduced earlier in the doc or even woven throughout in examples. Perhaps the first step of "Create a New Branch off the develop Branch" should be "get the latest from upstream develop".

An alternative to the "clone your fork" approach could be called "clone upstream". That's what all the core developers do. Since we have write access, we just push branches directly to the main repo. Contributors could use the "clone upstream" approach as well but they would need to push branches to their forks. I've noticed that @qqmyers will sometimes push branches to the QDR fork and sometimes to the GDCC fork. I sometimes push branches to by "pdurbin" fork but usually this is just experimental code. The advantage of the "clone upstream" approach is that to update develop you simply check it out and do a `git pull`. In contrast, these commands are more complicated in the "clone your fork" approach:

```
git fetch upstream develop
git rebase upstream/develop
```

(Incidentally, `git merge upstream/develop` (instead of `rebase`) should work fine and is the documented approach at https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork . I do see that both rebase and merge are presented at https://stackoverflow.com/a/7244456/19464 . If we remove "rebase" here it might help clarify that we don't expect contributors to rebase. We expect them to merge.)

One thing I'm not sure about in this "Summary of Git commands" section is why we bother documenting how to update the remote "develop" branch for one's fork. The "develop" branch of my fork at https://github.com/pdurbin/dataverse is thousands of commits behind upstream's "develop" but I don't think it does any harm. We could shorted the docs a bit by omitting this step.

I'm not sure if Péter is aware of this but the "How to Resolve Conflicts in Your Pull Request" section talks about updating from develop:

> The Dataverse Project team may ping you to ask you to merge the latest from the develop branch into your branch and resolve merge conflicts. If this sounds daunting, please just say so and we will assist you.

Then it goes on to give instructions for GitHub Desktop rather than the command line. Perhaps we can mention the new instructions being added in this pull request under that section.

In summary:

- This pull request is fine and fills in a gap in the docs (how to keep your pull request up to date with develop using the command line). Thank you, Péter!
- The core devs use a different process ("clone upstream") than what's documented ("clone your fork"). We could consider aligning these.
- I'm going to send this back to Péter to fix the typos and read the above. I'll also encourage the core team to read this.

**Comment by pkiraly** [comment]
@pdurbin My motivation was twofold. 1) I do not use Github Desktop 2) I was asked multiple times to update my develop branch during PRs are standing in the queue for a long time, and I always had to search for the steps (I wish I have a better memory). I am not a git expert, so if you provide more appropriate commands / workflows, I am more than happy with them.
All what I want is to just to copy and paste the git commands next time I am working with a pull request (with substitution of the provided bRanch name, issue number and commit message).

**Review by pdurbin — APPROVED** [review]
Looks good as of 236a25c.

**Comment by pdurbin** [comment]
@pkiraly thanks. That makes sense. This pull request definitely adds value so I went ahead and approved it. In the future, we can try to work out if a different method makes sense. Thanks!
